memmrg.h

```diff
@@ -0,0 +1,44 @@
+#ifndef MEMMRG_H
+#define MEMMRG_H
+
+
+typedef int Data;
+
+class MemMrg
+{
+    struct MrgItem
+    {
+        Data *item_val;
+        bool item_used;
+
+        MrgItem():item_used(false)
+        {
+          item_val = new Data;
```

Comment:
Have you forgotten that we are on a strange platform that does not allow us to allocate smal peases of memory?